text/0000-multi-part-examples-in-rustdoc/index.md

```diff
@@ -0,0 +1,231 @@
+
+An alternative to this is that specifying the language in the `split-start` block suffices and rustdoc can understand that all following parts of the multi-part example are of the same language.
+There is a problem with that.
+Pure markdown doesn't know and will never know anything about this feature.
+Markdown only [acknowledges](commonmark-fenced) the _typical_ use of the first word to specify a language:
+
+> The first word of the info string is typically used to specify the language of the code sample, and rendered in the class attribute of the code tag. However, this spec does not mandate any particular treatment of the info string.
+
+So any tools that are not rust-specific (or that don't yet implement multi-part examples) would not have the same understanding of this feature as rustdoc will, and therefore will think that only the first code block is whatever language was specified and the rest of the code blocks are not necessarily the same language.
+
+Erroring when the language tag varies between parts of a multi-part example prevents this situation.
+
+At the very least, it would probably prevent some authors from a future moment where they would think to themselves "oh, darn — I wish _I had_ placed the `ruby` tag on _every part_ of the multi-part example".
+
+Since the majority of multi-part examples are expected to be rust code, where the `rust` langauge tag is usually omitted, this error is not expected to be common.
+
+## Alternative design: rendered markdown_comments
+
+Example:
+
+    ```markdown_comments
+    let mut a = 3;
+    // Some _markdown_ [features](https://example.com/markdown_features).
+    a = 4;
+    dbg(a)
+    ```
+
+Rendered into something like this:
+
+![Example nested rendered alternative](./alternative_nested_rendered.png)
+
+The proposed `markdown_comments` tag would enable the feature in a code block.
+Then, any comments inside that block would be treated as markdown and rendered as HTML inside a container that is nested inside the code block, between the lines of code, where it is placed. These comments would no longer appear as comments inside the code, but as a block element with clear visual cues separating it from code.
+Rendered, they will not begin with `//`, etc.
+
+Drawbacks:
+
+- Already, a rust code example is typically a nested embedding:
+  a rust file; in which there's a markdown comment; in which there's a rust code block.
+  This proposal would add another layer of nesting to this.
+  Another layer of nesting would incur mental overhead and increased implementation cost for tooling compared with the suggested approach.
+- For longer nested comments, might be cumbersome.
+  Imagine comments that include code blocks. That's another nesting. Imagine long text. You'd have to handle line breaks and comment slashes (`//`).
```

Comment:
```suggestion
Drawbacks:

- Already, a rust code example is typically a nested embedding:
  a rust file; in which there's a markdown comment; in which there's a rust code block.
  This proposal would add another layer of nesting to this.
  Another layer of nesting would incur mental overhead and increased implementation cost for tooling compared with the suggested approach.
- For longer nested comments, might be cumbersome.
  Imagine comments that include code blocks. That's another nesting. Imagine long text. You'd have to handle line breaks and comment slashes (`//`).
- This would be very difficult to implement.
```

Reply:
Thank you. Added in c6346d3ee08e656735d0d6f2199a5b4828530539.